treeherder/webapp/api/intermittents_view.py

```diff
@@ -0,0 +1,124 @@
+                                            job__failure_classification__id=4
+                                            ).select_related('push').values('bug_id').annotate(
+                                            bug_count=Count('job_id')).values('bug_id', 'bug_count').order_by(
+                                            '-bug_count')
+        return queryset
+
+
+class FailuresByBug(generics.ListAPIView):
+    """ List of intermittent failure job details by bug, date range and tree (project name) """
+
+    serializer_class = FailuresByBugSerializer
+    pagination_class = CustomPagePagination
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
+        endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
+        tree = list(get_tree(self.request.query_params.get('tree').encode('utf-8')))
+        bug_id = int(self.request.query_params.get('bug'))
+
+        queryset = BugJobMap.objects.filter(bug_id=bug_id,
+                                            job__repository_id__in=tree,
+                                            job__push__time__range=(startday, endday)
+                                            ).select_related('job', 'push').values(
+                                            'bug_id', 'job_id', 'job__push__time', 'job__machine_platform__platform',
+                                            'job__repository__name', 'job__push__revision',
+                                            'job__signature__job_type_name', 'job__option_collection_hash',
+                                             ).order_by('-job__push__time')
+
+        hash_list = []
+
+        for item in queryset:
+            test_type = item['job__signature__job_type_name']
+            item['test_suite'] = re.sub(r'.+/', '', test_type)
+            match = filter(lambda x: item['job__option_collection_hash'] == x, hash_list)
+            if len(match) == 0:
+                hash_list.append(item['job__option_collection_hash'])
+
+        hash_query = OptionCollection.objects.filter(option_collection_hash__in=hash_list).select_related(
+                                                     'option').values('option__name', 'option_collection_hash')
+
+        for item in queryset:
+            match = filter(lambda x: item['job__option_collection_hash'] == x['option_collection_hash'], hash_query)
+            if len(match) > 0:
+                item['build_type'] = match[0]['option__name']
+            else:
+                item['build_type'] = 'unknown'
+
+        return queryset
+
+
+class FailureCount(generics.ListAPIView):
+    """ List of failures (optionally by bug) and testruns by day per date range and tree"""
+
+    serializer_class = FailureCount
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
```

Comment:
I need to concatenate the timestamp so as far as I know, I need to encode the params for that to happen (to turn a string into a datetime object). Since the default for a date w/o a timestamp is 00:00:00 I can leave that off of startday (and remove it from tree).